SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStatics);
+
+bool USpatialStatics::SpatialNetworkingEnabled()
+{
+	return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+bool USpatialStatics::SpatialOffloadingEnabled()
+{
+	return SpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::ActorGroupAssociationMatchesWorkerType(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		return false;
+	}
+
+	UActorGroupManager* ActorGroupManager = UActorGroupManager::GetInstance();
+
+	if (UWorld* World = Actor->GetWorld())
+	{
+		FString WorkerType = World->GetGameInstance()->GetSpatialWorkerType();
+
+		FName ActorGroup = ActorGroupManager->GetActorGroupForClass(Actor->GetClass());
+		if (ActorGroup == NAME_None) {
+			UE_LOG(LogSpatialStatics, Error, TEXT("Could not find actor group for actor: %s"), *Actor->GetName());
+			return false;
+		}
+
+		FString ActorWorkerAssociation = ActorGroupManager->GetWorkerTypeForClass(Actor->GetClass());
+		if (ActorWorkerAssociation.Equals(TEXT(""))) {
+			UE_LOG(LogSpatialStatics, Error, TEXT("Could not find worker association for actor group %s"), *ActorGroup.ToString());
+			return false;
+		}
+
+		return ActorWorkerAssociation.Equals(WorkerType);
+	}
+
+	UE_LOG(LogSpatialStatics, Error, TEXT("World was nullptr in USpatialStatics::CanHaveAuthority"));
+	return false;
+}
+
+bool USpatialStatics::IsGameServer(const UWorld* World)
+{
+	return World->GetGameInstance()->GetSpatialWorkerType().Equals(SpatialConstants::ServerWorkerType);
```

Comment:
null checks?

Reply:
Yea my bad, should just return false in case of UWorld* being nullptr, what do you think?

Reply:
Yes, `return World && ...` is what I would probably write

Reply:
I would also say you should log a warning if UWorld is nullptr. It's common in blueprints to do a IsValid check before passing an object somewhere.